src/server/scripts/EasternKingdoms/ScarletEnclave/chapter5.cpp

```diff
@@ -1664,8 +1666,37 @@ class npc_the_lich_king_tirion_dawn : public CreatureScript
 
 };
 
+// 58418 - Portal to Orgrimmar
+// 58420 - Portal to Stormwind
+class spell_teleport_leaders_blessing : public SpellScript
+{
+    PrepareSpellScript(spell_teleport_leaders_blessing);
+
+    bool Load() override
+    {
+        return GetCaster()->GetTypeId() == TYPEID_PLAYER;
+    }
+
+    void HandleScriptEffect(SpellEffIndex /* effIndex */)
+    {
+        Player* target = GetHitPlayer();
```

Comment:
The old code checks if the target is a player while the new code checks if the caster is a player. Maybe it's better to add a NULL check here too and return if target is null.

Reply:
I removed caster check since script anyway uses target only.